### ISSUE TITLE
Feature: Lecture term based course navigation

### ISSUE THREAD

**Issue by abias** [moodle-an-hochschulen/moodle-theme_boost_union#10]
Moodle core handles courses based on start and end dates of courses, nothing else. Something like lecture term based course navigation does not exist natively, but can be realized to some extent by using custom course fields and dashboard course list filters.

However, several institutions have the need to inject lecture based course navigation much deeper in Moodle course navigation.

This issue is a place to discuss the needs and to identify real pain points which could then be solved afterwards.
This issue also has a large overlap with the thoughts what https://moodle.org/plugins/local_boostnavigation does and should be doing on Moodle 4.x

**Comment by AkaiRingo** [comment]
Hey everyone :)
I would like to start the discussion for this issue based on the requirements we had for our implementation at Hochschule Hannover.
Maybe we can determine which of these requirements are needed for the average institution and can define a way of implementation for this issue.

Requirements:
1. Should replace the normal navigation entries
2. SETTING: Lecture terms should be limited to N entries (only show 5 lecture terms for example)
3. Navigation dropdowns/accordion for each lecture term
    - First entry is always open
    - Lecture Term entries are sorted based on the year/time
        - SETTING: There should be a setting to switch from ASC to DESC sorting
    - Each entry contains the lectures for the term
        - SETTING: Lectures are sorted based on short name (ASC/DESC)
        - Lectures should be highlighted when selected/active
4. An entry for lectures that are not related to a specific term (informational courses)
    - SETTING: A setting to set the position of this entry (before normal lecture terms or after)

Question:
1. How to handle many course entries in a lecture term - like above 15-20 entries?
    - Display all of them?
    - Add a “more” button after 10 entries?
        - It should not be another dropdown/accordion (too much nesting?)
    - Is this even relevant?

**Comment by abias** [comment]
Hi @AkaiRingo ,

thank you for starting the discussion for this goal here.

First, I would like to highlight that there is already a possibility to enable lecture term based course filtering on the My Course (my/courses.php) page by configuring a custom filter with the custom course field which holds the course's lecture term.

However, I understand you in a way that you do not want the student to go to the My Courses page everytime he wants to go to another course, you want to allow him to jump directly from one course to another (like it was possible before Moodle 4.x) and the presented course list should especially be clustered into the lecture terms.

Do you already have an idea how this course list should be presented to the end user? Could you provide some mockups or even Figma prototypes?

**Comment by AkaiRingo** [comment]
We discussed the topic again and looked at what moodle 4.0 already has. 
To give a faster access we would like to place a "Lecture Terms" navigation entry into the navigation bar to make it a "quick access". 
As you already said. We could maybe use the filtering in "my courses" for the implementation. 
Just for reference: https://tracker.moodle.org/browse/MDL-66017

We could also make it more generic by maybe allowing the institutions or even the users to decide which custom course field they want inside the "quick access".

Figma Prototype:
I used the Account Menu as a reference for the nested Menu in "Lecture Terms"
https://www.figma.com/proto/jsjEv0PLTp74Q95eOt3OUB/Semesternavigation?node-id=0%3A3&scaling=min-zoom&page-id=0%3A1&starting-point-node-id=0%3A3
